tooling/v8-snapshot/src/blueprint/custom-require.js

```diff
@@ -294,18 +294,22 @@ function createResolveOpts (relFilename, relDirname) {
  * @param relDirname dirname or module to resolve relative to project root
  * @returns result of invoking `require.resolve` with enhanced info
  */
-customRequire.resolve = function (mod, relFilename, relDirname) {
+customRequire.resolve = function (mod, ...args) {
   try {
+    // Handle the case where args is { paths: string[] }. The module is expected to be outside of the cypress snapshot.
+    if (args.length === 0 || (args[0] != null && typeof args[0] !== 'string')) {
+      return require.resolve(mod, ...args)
+    }
+
+    const relFilename = args.length > 0 ? args[0] : null
+    const relDirname = args.length > 1 ? args[1] : null
```

Comment:
nit: don't need length checks here, you'll just get 'undefined' for out of bounds lookups

Reply:
Good call [cypress-io/cypress@`ec34a53` (#23732)](https://github.com/cypress-io/cypress/pull/23732/commits/ec34a537a4879edcfbc0409b0f81f16a94e97ee9)